src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1618,654 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+                me->GetMotionMaster()->MovePoint(POINT_CLOSE_TO_ZONE, LordaeronFinishPos, false, 3.5f);
+        }
+    }
+};
+
+enum SylvanasWindrunnerLordaeron
+{
+    SPELL_DREADGUARD_SALUTE_AURA                = 84200,
+    SPELL_KILL_ME                               = 84180,
+
+    EVENT_SYLVANAS_RIDE_WARHORSE_LORDAERON      = 1,
+    EVENT_SYLVANAS_HEARTSTRIKE_COOLDOWN         = 3,
+    EVENT_SYLVANAS_TALK                         = 4,
+    EVENT_FINISH_SCENE_LORDAERON                = 26,
+
+    TALK_SYLVANAS_LORDAERON_0                   = 0,
+    TALK_SYLVANAS_LORDAERON_1                   = 1,
+    TALK_SYLVANAS_LORDAERON_2                   = 2,
+    TALK_SYLVANAS_LORDAERON_3                   = 3,
+    TALK_SYLVANAS_LORDAERON_4                   = 4,
+    TALK_SYLVANAS_LORDAERON_5                   = 5,
+    TALK_SYLVANAS_LORDAERON_6                   = 6,
+    TALK_SYLVANAS_LORDAERON_7                   = 7,
+    TALK_SYLVANAS_LORDAERON_8                   = 8,
+    TALK_SYLVANAS_LORDAERON_9                   = 9,
+    TALK_SYLVANAS_LORDAERON_10                  = 10,
+    TALK_SYLVANAS_LORDAERON_11                  = 11,
+    TALK_SYLVANAS_LORDAERON_12                  = 12,
+    TALK_SYLVANAS_LORDAERON_13                  = 13,
+    TALK_SYLVANAS_LORDAERON_14                  = 14,
+    TALK_SYLVANAS_LORDAERON_15                  = 15,
+    TALK_SYLVANAS_LORDAERON_16                  = 16,
+    TALK_SYLVANAS_LORDAERON_17                  = 17,
+    TALK_SYLVANAS_LORDAERON_18                  = 18,
+    TALK_SYLVANAS_LORDAERON_19                  = 19,
+    TALK_SYLVANAS_LORDAERON_20                  = 20,
+    TALK_SYLVANAS_LORDAERON_21                  = 21
+};
+
+// 45051 - Lady Sylvanas Windrunner (Lordaeron)
+struct npc_silverpine_sylvanas_lordaeron : public ScriptedAI
+{
+    npc_silverpine_sylvanas_lordaeron(Creature* creature) : ScriptedAI(creature), _heartstrikeCD(false) { }
+
+    void JustAppeared() override
+    {
+        if (Creature* sylvanasHorse = me->FindNearestCreature(NPC_FORSAKEN_WARHORSE_SYLVANAS, 25.0f))
+        {
+            _sylvanasHorseGUID = sylvanasHorse->GetGUID();
+
+            me->EnterVehicle(sylvanasHorse, SEAT_WARHORSE_SYLVANAS);
+
+            _events.ScheduleEvent(EVENT_SYLVANAS_RIDE_WARHORSE_LORDAERON, 500ms);
+        }
+
+        DoCastSelf(SPELL_DREADGUARD_SALUTE_AURA);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        _playerGUID = summoner->GetGUID();
+    }
+
+    void SpellHit(WorldObject* caster, SpellInfo const* spellInfo) override
+    {
+        switch (spellInfo->Id)
+        {
+            case SPELL_KILL_ME:
+            {
+                if (_heartstrikeCD || me->HasUnitState(UNIT_STATE_CASTING))
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_NONE);
+                else
+                {
+                    me->HandleEmoteCommand(EMOTE_STATE_HOLD_BOW);
+
+                    if (caster->ToUnit()->GetEntry() == NPC_WORGEN_RENEGADE)
```

Comment:
`ToUnit()` cast is not required; `GetEntry()` is part of `WorldObject`

Reply:
Feedback applied, in 434 branch GetEntry was only part of Unit/Creature.